java/org/tartarus/snowball/SnowballProgram.java

```diff
@@ -303,57 +346,43 @@ protected void slice_check()
     {
 	if (bra < 0 ||
 	    bra > ket ||
-	    ket > limit ||
-	    limit > current.length())   // this line could be removed
+	    ket > limit)
```

Comment:
Maybe we should restore the "could be removed" check which would now be: `limit > length`

I think the point the comment tries to make is that `limit > length` should be impossible in a properly functioning system, so maybe an assertion is more appropriate.

Reply:
I changed these to java `assert`s and enabled assertions in the Makefile when running tests.